sdk/zklogin/src/checks.ts

```diff
@@ -0,0 +1,282 @@
+	 * Do not use it elsewhere unless you know what you're doing.
+	 */
+	getRawClaimValue(name: string): string {
+		if (!Object.prototype.hasOwnProperty.call(this.processed, name)) {
+			throw new Error('Claim ' + name + ' not processed');
+		}
+		const details = this.processed[name];
+
+		const value_index = details.offsets.value + details.offsets.start;
+		const value_length = details.offsets.value_length;
+		if (this.decoded_payload[value_index] !== '"') {
+			throw new Error(
+				`Claim ${name} does not have a string value. Details: ${JSON.stringify(details)}`,
+			);
+		}
+		if (this.decoded_payload[value_index + value_length - 1] !== '"') {
+			throw new Error(
+				`Claim ${name} does not have a string value. Details: ${JSON.stringify(details)}`,
+			);
+		}
+
+		const raw_value = this.decoded_payload.slice(value_index + 1, value_index + value_length - 1); // omit the quotes
+		if (raw_value !== details.value) {
+            // TODO: This is an interesting event to note.
+            //       Note sure if the console.info is the right way to log it.
+			console.info(
+				`Claim value ${raw_value} of length ${
+					raw_value.length
+				} has escapes. Details: ${JSON.stringify(details)}`,
+			);
+		}
+		return raw_value;
+	}
+}
+
+export function lengthChecks(header: string, payload: string, keyClaimName: string, processor: JSONProcessor) {
+	/// Is the header length small enough
+	const header_len = header.length;
+	if (header_len > MAX_HEADER_LEN_B64) {
+		throw new Error(`Header is too long`);
+	}
+
+	/// Is the combined length of the header and payload small enough
+	const unsigned_jwt = header + '.' + payload;
+	const L = unsigned_jwt.length * 8;
+	const K = (512 + 448 - ((L % 512) + 1)) % 512;
+	if ((L + 1 + K + 64) % 512 !== 0) {
+		throw new Error('This should never happen');
+	}
+
+	// The SHA2 padding is 1 followed by K zeros, followed by the length of the message
+	const padded_unsigned_jwt_len = (L + 1 + K + 64) / 8;
+
+	// The padded unsigned JWT must be less than the max_padded_unsigned_jwt_len
+	if (padded_unsigned_jwt_len > MAX_PADDED_UNSIGNED_JWT_LEN) {
+		throw new Error(`The JWT is too long`);
+	}
+
+	const keyClaimDetails = processor.process(keyClaimName); // throws an error if key claim name is not found
+	const keyClaimValue = processor.getRawClaimValue(keyClaimName);
+	const keyClaimValueLen = keyClaimValue.length;
+	if (keyClaimValueLen > MAX_KEY_CLAIM_VALUE_LENGTH) {
+		throw new Error('Key claim value is too long');
+	}
+    // Note: Key claim name length is being checked in genAddressSeed.
+
+	/// Are the extended claims small enough (key claim, email_verified)
+	const extendedKeyClaimLen = keyClaimDetails.ext_claim.length;
+	if (extendedKeyClaimLen > MAX_EXTENDED_KEY_CLAIM_LEN) {
+		throw new Error(`Extended key claim length is too long`);
+	}
+
+	if (keyClaimName === 'email') {
+		const evClaimDetails = processor.process('email_verified');
```

Comment:
I think FB does not return `email_verified`

Reply:
I see. FYI, `email_verified` is a must if the key claim name is email. That probably means we can't use email with FB.